### PR TITLE
Benchmark refactoring, improving efficiency, enhanced reporting

### PR DESCRIPTION
**Pull request by ibnesayeed**
* Refactored benchmark code to reduce the number of reads of the data file
* Added SpecReporter that shows each test case name grouped with suit names instead of the default progress bar that was not very informative
* Added custom reporter for benchmarks to better represent the information in a copy paste friendly format -- bye bye `NOPROGRESS` ENV hack
* Added record counts on which benchmarks were performed

### CODE REVIEW

**Comment by Ch4s3** [comment]
I'll take a look later today, thanks!

**Comment by ibnesayeed** [comment]
> @Ch4s3: I'll take a look later today, thanks!

I am making some changes here to better organize the code.

**Comment by ibnesayeed** [comment]
@Ch4s3, I think I am done refactoring for now. Please go ahead for code review and merging if seems good.

File: test/bayes/bayesian_common_benchmarks.rb [hunk 1]
```diff
@@ -1,9 +1,28 @@
 # encoding: utf-8
 
 module BayesianCommonBenchmarks
-  def load_data
-    sms_spam_collection = File.expand_path(File.dirname(__FILE__) + '/../data/corpus/SMSSpamCollection.tsv')
-    File.read(sms_spam_collection).force_encoding("utf-8").split("\n")
+  MAX_RECORDS = 5000
+
+  class BenchmarkReporter < Minitest::Reporters::RubyMateReporter
```
Comment (ibnesayeed) [inline test/bayes/bayesian_common_benchmarks.rb]: I would have much preferred to extract this custom reporter into a separate file and potentially share that with other benchmarks in future, but the kind of data I am using in the `before_suite` method will not be available outside, unless there is a way that I am unaware of.

File: test/bayes/bayesian_common_benchmarks.rb [hunk 2]
```diff
@@ -1,9 +1,28 @@
...
+    def before_suite(suite)
+      puts ([suite] + BayesianCommonBenchmarks.bench_range).join("\t")
```
Comment (ibnesayeed) [inline test/bayes/bayesian_common_benchmarks.rb]: I would have much preferred to extract this custom reporter into a separate file and potentially share that with other benchmarks in future, but the kind of data I am using in the `before_suite` method will not be available outside, unless there is a way that I am unaware of.

Comment (Ch4s3) [inline test/bayes/bayesian_common_benchmarks.rb]: This is probably fine

**Pull request merged**